### ISSUE TITLE
Shuffle and skip Button for youtube playlists and account linking (maybe?)

### ISSUE THREAD

**Issue by CoolQMan** [EuCaue/gnome-shell-extension-quick-lofi#3]
Hey, I stumbled across this extension 2 days ago and I am loving it. 
I just tried to put in a youtube playlist link and it surprizingly worked!! 

I was wondering if it was possible to add a "Shuffle" button which would shuffle the music in the playlist when toggled and a "skip" button which would skip the current track and also a pause button to pause the current song (these all would work only in youtube playlists obviously)

I don't know if this works for spotify (I don't use spotify) but if it does then these features would be great for spotify users as well! 

And about the account linking part, I know this is a very big request so maybe keep this on low priority but would it be possible to link your google account to this extension and then it would pull all the pre-made playlists and the Youtube mixes (the auto generated playlists which youtube recommends) which we would be able to pin and play 
I know this is a huge request but if you do find the time to implement this it would be amazing! 

Thanks and once again amazing extension!!

Edit: I forgot to mention one thing. Whenever turning off the extension or switching between radio if there was a fade out transition then it would sound amazing.

**Comment by EuCaue** [comment]
Hi! :) 

<small>Sorry for the late response</small>

Glad that you're liking the extension. ❤️ 

> I was wondering if it was possible to add a "Shuffle" button which would shuffle the music in the playlist when toggled, and a "skip" button which would skip the current track, as well as a pause button to pause the current song (these would only work with YouTube playlists, obviously).

I think it's possible to do that. I'll just have to see how to get the necessary information from `MPV` to update the **UI** correctly. And that is a really nice idea; I do plan to make a "little" interface on right-click to control the volume and things like that.

> I don't know if this works for Spotify (I don't use Spotify), but if it does, then these features would be great for Spotify users as well!

Well, I don't think so. This extension uses `MPV` with `yt-dlp` to play YouTube playlists/videos. I'm almost sure it does not support Spotify, but it could be a feature in the future if there's something to play Spotify playlists/songs. The sad part is that most third-party Spotify apps require a Spotify Premium account.

> And about the account linking part, I know this is a very big request, so maybe keep this on low priority, but would it be possible to link your Google account to this extension, and then it would pull all the pre-made playlists and the YouTube mixes (the auto-generated playlists that YouTube recommends) that we could pin and play? I know this is a huge request, but if you do find the time to implement this, it would be amazing!

_Yep_, that's definitely a big request, but I do think it's possible. Although I'm a little bit concerned if I'm going to need a YouTube **API** key to use that or if I can use the GNOME account system with the Google account option, that would be great. This is a really nice idea, but I will study what's necessary in the future.

Thank you for your support and ideas! 😄 

> Edit: I forgot to mention one thing. Whenever turning off the extension or switching between radio stations, if there was a fade-out transition, it would sound amazing.

Hm, that's an interesting one. I'll see if it's possible, but I'll definitely keep it on my to-do list!

**Comment by CoolQMan** [comment]
Thanks for the reply.

I tried to a spotify playlist earlier and yeah it didn't work sadly. 

A right click menu would be awesome to implement that. Like it can have a play/pause button, prev track, next track, a slider to peek, a toggle for shuffle and thumbnail. And the below that there can be the volume control slider (or maybe a vertical slider to the right side of the mini-player). 
And I think to access the current right click menu (where you add the radio stations etc) can still be accessed by doing a middle click. 
This thing alone would make this brilliant extension even better.

**Comment by EuCaue** [comment]
> I tried to a spotify playlist earlier and yeah it didn't work sadly.

Yeah, unfortunately, it looks like Spotify won't be an option—at least not without a premium account and an API key. :/

> A right click menu would be awesome to implement that. Like it can have a play/pause button, prev track, next track, a slider to peek, a toggle for shuffle and thumbnail. And the below that there can be the volume control slider (or maybe a vertical slider to the right side of the mini-player). And I think to access the current right click menu (where you add the radio stations etc) can still be accessed by doing a middle click. This thing alone would make this brilliant extension even better.

A mini-player would be awesome, but I think it’d be most useful with YouTube playlists. Otherwise, a lot of the buttons might end up being disabled, right?

There’s also the issue of not having a thumbnail to show, so I’ll need to figure out a way to keep everything looking good. A dynamic UI that adjusts based on what’s playing would be ideal—like disabling the prev/next track buttons when you’re not using a playlist. Plus, I’m not sure how peeking at a live stream would work, if there are any drawbacks.

The biggest challenges here are the UI design and figuring out what kind of info yt-dlp provides to handle all the buttons, along with any restrictions GTK might impose. I’ll definitely play around with this idea in the future. For now, I’m thinking of adding some sort of volume control that doesn’t require going into the settings. :smile:

By the way, do all your radio stations come from YouTube, or do you use other sources too?

**Comment by CoolQMan** [comment]
> By the way, do all your radio stations come from YouTube, or do you use other sources too?

Don't know many radio stations so I just use youtube playlists.
